### PR TITLE
Feature/transform point

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [ ] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

## Description

add transform point function to the math namespace.

## How to review this PR.

Please check passing all CI test cases.

### CODE REVIEW

File: simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp [hunk 1]
```diff
@@ -23,6 +23,10 @@ namespace math
 {
 const geometry_msgs::msg::Pose getRelativePose(
   const geometry_msgs::msg::Pose & from, const geometry_msgs::msg::Pose & to);
+const geometry_msgs::msg::Point transformPoint(
+  const geometry_msgs::msg::Pose & pose, const geometry_msgs::msg::Point & point);
+std::vector<geometry_msgs::msg::Point> transformPoints(
+  geometry_msgs::msg::Pose pose, std::vector<geometry_msgs::msg::Point> points);
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp:29]:
```
std::vector<geometry_msgs::msg::Point> transformPoints(
  geometry_msgs::msg::Pose pose, std::vector<geometry_msgs::msg::Point> points);
```
↓
```
std::vector<geometry_msgs::msg::Point> transformPoints(
  const geometry_msgs::msg::Pose & pose, const std::vector<geometry_msgs::msg::Point> & points);
```

File: simulation/traffic_simulator/src/math/transform.cpp [hunk 2]
```diff
@@ -59,5 +60,34 @@ const geometry_msgs::msg::Pose getRelativePose(
...
+  v(2) = v(2) + pose.position.z;
+  geometry_msgs::msg::Point transformed;
+  transformed.x = v(0);
+  transformed.y = v(1);
+  transformed.z = v(2);
+  return transformed;
+}
+
+std::vector<geometry_msgs::msg::Point> transformPoints(
+  geometry_msgs::msg::Pose pose, std::vector<geometry_msgs::msg::Point> points)
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/transform.cpp:84]:
```
std::vector<geometry_msgs::msg::Point> transformPoints(
  geometry_msgs::msg::Pose pose, std::vector<geometry_msgs::msg::Point> points);
```
↓
```
std::vector<geometry_msgs::msg::Point> transformPoints(
  const geometry_msgs::msg::Pose & pose, const std::vector<geometry_msgs::msg::Point> & points);
```

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: simulation/traffic_simulator/src/math/transform.cpp (hunk 2 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/transform.cpp]: Fixed in https://github.com/tier4/scenario_simulator_v2/pull/633/commits/8a484b8ff81788c5188b6d779e81b84a87eff5be

File: simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp]: Fixed in https://github.com/tier4/scenario_simulator_v2/pull/633/commits/8a484b8ff81788c5188b6d779e81b84a87eff5be

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**